controllers/hvpa_controller_test.go

```diff
@@ -191,6 +192,24 @@ var _ = Describe("#TestReconcile", func() {
 				return fmt.Errorf("Error: Expected 1 HPA; found %v", len(hpaList.Items))
 			}, timeout).Should(Succeed())
 
+			// Update the HPA spec and expect the change to be reconciled
+			hvpa := &autoscalingv1alpha1.Hvpa{}
+			c.Get(context.TODO(), types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, hvpa)
+			hvpa.Spec.Hpa.Template.Spec.MinReplicas = pointer.Int32(3)
+			err = c.Update(context.TODO(), hvpa)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(func() error {
+				var actualMinReplicas int32
+				c.List(context.TODO(), hpaList)
+				for _, hpa := range hpaList.Items {
+					actualMinReplicas = *hpa.Spec.MinReplicas
+				}
+				if actualMinReplicas != 3 {
```

Comment:
I assume that you are intentionally not differentiating the case "minReplicas != 3" from the case "no HPA deployed".

Reply:
Earlier in the test it is verified that there is an HPA deployed. This part is just to verify that the minReplicas have been changed as expected. So, yes, I think we don't need to treat this case separately.